core/spt.api/src/main/java/mb/spt/api/analyze/TestableAnalysis.java

```diff
@@ -0,0 +1,17 @@
+package mb.spt.api.analyze;
+
+import mb.common.option.Option;
+import mb.common.region.Region;
+import mb.common.result.Result;
+import mb.common.util.ListView;
+import mb.pie.api.Session;
+import mb.resource.ResourceKey;
+import mb.resource.hierarchical.ResourcePath;
+import org.checkerframework.checker.nullness.qual.Nullable;
+import org.spoofax.interpreter.terms.IStrategoAppl;
+import org.spoofax.interpreter.terms.IStrategoTerm;
+
+public interface TestableAnalysis {
+    Result<IStrategoTerm, ?> testRunStrategy(Session session, ResourceKey resource, String strategy, Option<ListView<RunArgument>> arguments, Option<Region> region, @Nullable ResourcePath rootDirectoryHint) throws InterruptedException;
```

Comment:
Related to what I mentioned [in this comment](https://github.com/metaborg/spoofax-pie/pull/24#discussion_r683205404), the arguments can contain a region which has to be converted to a node in the AST in the `TestStrategoTaskDef`. Although I could modify `StrategoRunArgument` to be either an `IStrategoTerm` or a `Region` (currently it's either an `IStrategoInt`, `IStrategoString` or `Region`, since those are the only arguments that can be provided from an SPT test)

Reply:
That is fine, we can always change it later.